ngrinder-controller/src/main/java/org/ngrinder/agent/model/NGrinderPackage.java

```diff
@@ -0,0 +1,219 @@
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.commons.lang.StringUtils.isNotEmpty;
+import static org.apache.commons.lang.StringUtils.trimToEmpty;
+import static org.ngrinder.common.util.CollectionUtils.buildMap;
+import static org.ngrinder.common.util.CollectionUtils.newHashMap;
+
+/**
+ * This enum is designed to support create packages.(agent, monitor)
+ *
+ * @since 3.5
+ */
+public enum NGrinderPackage {
+
+	AGENT(
+		"ngrinder-agent",
+		"ngrinder-agent/",
+		"ngrinder-agent/lib/",
+		"classpath*:ngrinder-sh/agent/*",
+		"agent_agent.conf",
+		"dependencies.txt"
+	),
+
+	MONITOR(
+		"ngrinder-monitor",
+		"ngrinder-monitor/",
+		"ngrinder-monitor/lib/",
+		"classpath*:ngrinder-sh/monitor/*",
+		"agent_monitor.conf",
+		"monitor-dependencies.txt"
+	);
+
+	public static final int TIME_MILLIS_OF_DAY = 1000 * 60 * 60 * 24;
+	private static Config config;
+	private Logger LOGGER = LoggerFactory.getLogger(Package.class);
+
+	private String moduleName;
+	private String basePath;
+	private String libPath;
+	private String shellScriptsPath;
+	private String templateName;
+	private String dependenciesFileName;
+
+	NGrinderPackage(String moduleName, String basePath, String libPath, String shellScriptsPath, String templateName, String dependenciesFileName) {
+		this.moduleName = moduleName;
+		this.basePath = basePath;
+		this.libPath = libPath;
+		this.shellScriptsPath = shellScriptsPath;
+		this.templateName = templateName;
+		this.dependenciesFileName = dependenciesFileName;
+	}
+
+	public String getModuleName() { return moduleName; }
+	public String getBasePath() { return basePath; }
+	public String getLibPath() { return libPath; }
+	public String getShellScriptsPath() { return shellScriptsPath; }
+	public String getTemplateName() { return templateName; }
+	public String getDependenciesFileName() { return dependenciesFileName; }
+
+	public Map<String, Object> getConfigParam(String regionName, String controllerIP, int port, String owner) {
+		switch (this) {
+			case AGENT:
+				return getAgentConfigParam(regionName, controllerIP, port, owner);
+			case MONITOR:
+				return buildMap("monitorPort", (Object) String.valueOf(port));
+			default:
+				throw new IllegalArgumentException(this.getModuleName() + " module is not supported");
+		}
+	}
+
+	public Set<String> getPackageDependentLibs(URLClassLoader urlClassLoader) {
+		Set<String> libs = getDependentLibs(urlClassLoader);
+		if (this.equals(AGENT)) {
+			libs.add("ngrinder-core");
+			libs.add("ngrinder-runtime");
+			libs.add("ngrinder-groovy");
+		}
+		return libs;
+	}
+
+	public File makePackageTar(String regionName, String connectionIP, String ownerName, boolean forWindow) {
+		File packageDir = getPackagesDir();
+		if (packageDir.mkdirs()) {
+			LOGGER.info("{} is created", packageDir.getPath());
+		}
+		final String packageName = getDistributionPackageName(regionName, connectionIP, ownerName, forWindow);
+		return new File(packageDir, packageName);
+	}
```

Comment:
I dont't think it behave differently from its name? is it really ```make``` package tar?

Reply:
Get tar archive output stream from this file. and add dependent libraries.
so finally created tar file is same this.

Reply:
if so, it's only respossible to create **File name** not File content.

I think it's better to name it getPackageFileName()